experimental/harfbuzz/src/lib.rs

```diff
@@ -180,31 +214,62 @@ struct ScriptDataForHarfBuzz {
 unsafe extern "C" fn icu4x_hb_unicode_script(
     _ufuncs: *mut hb_unicode_funcs_t,
     unicode: hb_codepoint_t,
-    user_data: *mut c_void,
+    #[cfg_attr(not(feature = "alloc"), allow(unused_variables))] user_data: *mut c_void,
 ) -> hb_script_t {
-    let script_data: &ScriptDataForHarfBuzz = &*(user_data as *mut ScriptDataForHarfBuzz);
-    let script: Script = script_data.script_map.as_borrowed().get32(unicode);
-    let enum_to_name_mapper = script_data.enum_to_name_mapper.as_borrowed();
-    let name: TinyStr4 = enum_to_name_mapper
+    #[cfg(feature = "compiled_data")]
+    const SCRIPT_MAP: Option<CodePointMapDataBorrowed<'static, Script>> = Some(maps::script());
+    #[cfg(feature = "compiled_data")]
+    const ENUM_TO_NAME_MAPPER: Option<PropertyEnumToValueNameLinearTiny4MapperBorrowed<Script>> =
+        Some(Script::enum_to_short_name_mapper());
+    #[cfg(not(feature = "compiled_data"))]
+    const SCRIPT_MAP: Option<CodePointMapDataBorrowed<'static, Script>> = None;
+    #[cfg(not(feature = "compiled_data"))]
+    const ENUM_TO_NAME_MAPPER: Option<PropertyEnumToValueNameLinearTiny4MapperBorrowed<Script>> =
+        None;
+
+    #[cfg(feature = "alloc")]
+    let script_map = (user_data as *mut ScriptDataForHarfBuzz)
+        .as_ref()
+        .map(|d| d.script_map.as_borrowed());
+    #[cfg(feature = "alloc")]
+    let enum_to_name_mapper = (user_data as *mut ScriptDataForHarfBuzz)
+        .as_ref()
+        .map(|d| d.enum_to_name_mapper.as_borrowed());
+    #[cfg(not(feature = "alloc"))]
+    let script_map = None;
+    #[cfg(not(feature = "alloc"))]
+    let enum_to_name_mapper = None;
+
+    let script = script_map.or(SCRIPT_MAP).unwrap_unchecked().get32(unicode);
+    let name = enum_to_name_mapper
+        .or(ENUM_TO_NAME_MAPPER)
+        .unwrap_unchecked()
         .get(script)
         .unwrap_or(tinystr!(4, "Zzzz"));
+
     hb_script_from_string(
         name.as_ptr() as *const c_char,
         name.len().try_into().unwrap_or(0),
     )
 }
```

Comment:
why are we going through names here instead of including a map from ICU4X `Script` to HB `hb_script_t`?

Reply:
`hb_script_t` is a `[u8; 4]` of the script code.